nats/bench_command.go

```diff
@@ -49,13 +65,35 @@ func configureBenchCommand(app *kingpin.Application) {
 	bench.Flag("size", "Size of the test messages").Default("128").IntVar(&c.msgSize)
 	bench.Flag("csv", "Save benchmark data to CSV file").StringVar(&c.csvFile)
 	bench.Flag("progress", "Enable progress bar while publishing").Default("true").BoolVar(&c.progress)
-	bench.Flag("ack", "Waits for acknowledgement on messages using Requests rather than Publish").Default("false").BoolVar(&c.ack)
+	bench.Flag("syncpub", "Synchronously waits for acknowledgement on published messages (using Request in NATS or synchronous Publish in JS)").Default("false").BoolVar(&c.syncPub)
+	bench.Flag("js", "Use JetStream streaming").Default("false").BoolVar(&c.js)
+	bench.Flag("pubbatch", "Sets the batch size for JS asynchronous publishing").Default("100").IntVar(&c.pubBatch)
+	bench.Flag("jsfile", "Persist the stream to file").Default("false").BoolVar(&c.jsFile)
+	bench.Flag("pull", "Uses a JS pull consumer").Default("false").BoolVar(&c.pull)
+	bench.Flag("pullbatch", "Sets the batch size for the JS pull consumer").Default("100").IntVar(&c.pullBatch)
+	bench.Flag("ack", "Acks consumption of messages").Default("false").BoolVar(&c.ack)
+	bench.Flag("replicas", "Number of stream replicas").Default("1").IntVar(&c.replicas)
+	bench.Flag("nopurge", "Do not purge the stream before running").Default("false").BoolVar(&c.noPurge)
+	bench.Flag("nodelete", "Do not delete the stream at the end of the run").Default("false").BoolVar(&c.noDeleteStream)
```

Comment:
I like how the tools will clean up after itself.  +1

Reply:
Boolean flags like `--purge` would automatically support `--no-purge` as a side effect, the help doesnt show them clearly so for these lets make them `--no-delete` and `--no-purge` for consistency. Also `--no-progress` and `--no-queue`.

Relying on the `--purge` and supporting `--no-purge` as side effect also avoid all these stupid go negative booleans, we have statements like `if !noPurge {...}` which is one of the worst things in go forcing people to do write double negatives all the time.  So had these variables been `purge` and default to `"true"` code all reads easier.  You can still have `--no-purge` if you like but call the variable `purge` and defaults to `"true"` in the CLI to avoid the double negatives

Reply:
As I mentioned elsewhere I changed to using --noFoo format because many people may not know about the --no-foo side effect to unset the flag (and because doing something like --foo=false doesn't work). I don't mind changing back to --foo but will then add a reminder of how to unset a flag set to true by default in the help.

Reply:
Please explain the difference between:

```
 	bench.Flag("nodelete", "Do not delete the stream at the end of the run").Default("false").BoolVar(&c.noDeleteStream) 
```

and


```
 	bench.Flag("no-delete", "Do not delete the stream at the end of the run").Default("true").BoolVar(&c.deleteStream) 
```

Reply:
(sorry, that was a bad example let me see where I did this correctly)

Reply:
There's a nasty way to work around this - so for now lets just use `bench.Flag("no-delete".....)` and not `bench.Flag("nodelete",....)` thus keeping the `--no-*` pattern consistent but keep the variable names as is.

Reply:
done